pkg/endpoint/endpoint.go

```diff
@@ -366,6 +370,11 @@ func (e *Endpoint) UpdateController(name string, params controller.ControllerPar
 	e.controllers.UpdateController(name, params)
 }
 
+// TriggerController triggers the endpoint's controller with the specified name.
+func (e *Endpoint) TriggerController(name string) {
+	e.controllers.Lookup(name).Trigger()
```

Comment:
Since this is an exported function, we need to check that `Lookup(name) != nil`.